mongo/mongo.go

```diff
@@ -153,21 +154,32 @@ func (m *Mongo) ListInteractives(ctx context.Context, offset, limit int, modelFi
 // generate filter string depending on data type
 // string eq value
 // array in value(s)
-func generateFilter(model *models.InteractiveMetadata) bson.M {
+func generateFilter(model *models.InteractiveFilter) bson.M {
 	filter := bson.M{}
 	filter["active"] = bson.M{"$eq": true}
-	if model == nil {
+	if model == nil || model.Metadata == nil {
```

Comment:
is this right? metadata might be nil if we use the other filter?

Reply:
metadata needs to be there always ... for both cases..
for linked-collection filter.. the collection-id needs to be in the metadata struct